tidb-external-ts.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Read Historical Data Using the `tidb_external_ts` Variable
+summary: Learn how to read historical data using the `tidb_external_ts` variable.
+---
+
+# Read Historical Data Using the `tidb_external_ts` Variable
+
+This document describes how to perform the [Stale Read](/stale-read.md) feature using the `tidb_external_ts` variable to read historical data in TiDB, including specific usage examples.
+
+> **Warning:**
+>
+> Currently, you cannot use Stale Read together with TiFlash. If your SQL query contains the `AS OF TIMESTAMP` clause and TiDB might read data from TiFlash replicas, you might encounter an error with a message like `ERROR 1105 (HY000): stale requests require tikv backend`.
+>
+> To fix the problem, disable TiFlash replicas for your Stale Read query. To do that, perform one of the following operations:
+>
+> - Use the `set session tidb_isolation_read_engines='tidb,tikv'` variable.
+> - Use the [hint](/optimizer-hints.md#read_from_storagetiflasht1_name--tl_name--tikvt2_name--tl_name-) to enforce TiDB to read data from TiKV.
+
+## Scenarios
+
+Read with an absolute timestamp is especially useful for data replication software (for example TiCDC). After the data replication software ensures that all data before a timestamp have been synchronized to the downstream, it can set the `tidb_external_ts` for the downstream TiDB, so that the read queries of the downstream TiDB can read consistent data.
+
+## Usage
+
+The `tidb_external_ts` variable is used to specify the timestamp of the historical data to be read.
+
+The `tidb_enable_external_ts_read` controls whether TiDB will read with the `tidb_external_ts` variable. The default value is `OFF`, which means that the `tidb_external_ts` variable is ignored. If `tidb_enable_external_ts_read` is set `ON` globally, all the queries will read historical data. If `tidb_enable_external_ts_read` is set `ON` for a session, only the queries in the session will read historical data.
+
+When `tidb_enable_external_ts_read` is enabled, TiDB becomes read-only. All the write queries will fail with an error like `ERROR 1836 (HY000): Running in read-only mode`.
```

Comment:
This line is not in chinese document too.

Reply:
Fixed.